M2/Macaulay2/d/interface.dd

```diff
@@ -66,8 +66,7 @@ export rawFareyApproximation(e:Expr):Expr := (
 setupfun("rawFareyApproximation", rawFareyApproximation);
 export rawRandomQQ(e:Expr):Expr := (
      when e
-     is Nothing do toExpr(Ccode(QQ, "rawRandomQQ(", "0)"))
-     is ht:ZZcell do toExpr(Ccode(QQ, "rawRandomQQ(", ht.v, ")"))
+     is ht:ZZcell do toExpr(Ccode(QQorNull, "rawRandomQQ(", ht.v, ")"))
```

Comment:
I think it might be simpler to just assert that Height is nonzero in the top-level, no?

Reply:
My concern was that we also want to catch this when making random matrices, which is a different call to the engine that isn't specific to the coefficient ring.  Is height always positive for all the various meanings of height we might want for `random`?  Or could it be zero/negative in other contexts?